code-challenges.test.js

```diff
@@ -23,24 +30,136 @@ const people = [
 
 // b) Create the function that makes the test pass.
 
+// PC: Input is an array, that contains objects. Output will be an array of strings, composed of data from the objects with data 
+//  the value from the key name will be capitalized, and inserted into a string, with the word "is" between the name value and the occupation value and a "." at the end of the string. String interpolation used to access data, 
+// How do I capitalize both names = .toUpperCase, but need to be able to access both names in string. 
+
+// What will allow me to do an action to each value in array: map over the array
+// Used .split to separate names in the string
+// Used another .map to iterate over each split name
+// value.charAt(0) -- selects only the character at the index of 0
+// .toUppercase capitalizes the character
+// + adds the capitalized character to the value that follows it
+// value.slice(1) cuts the value beginning at the index of 1 (discards the previous)
+// together this returns the sought after capitalized name. - CLOSE THE SECOND MAP
+// STILL WITHIN FIRST MAP: .join(" ") is used to rejoin the first and second name with a space in between.
+// + adds the first names to the string that follows: " is " - spaces included
+// + adds the first sections to 
+// input.occupation (this works because we are still within the first .map!) 
+// + adds the first sections to the final string: "."
+// and returns the expected output!
+
+const peopleCaps = (array) => {
+  return array.map(input =>
+      input.name
+      .split(" ")
+      .map(value =>
+          value.charAt(0)
+          .toUpperCase()
+          +value.slice(1))
+      .join(" ") + " is " + input.occupation + "."
+      )
+}
+
```

Comment:
Good work. You found a way to support the pseudocode. These notes will be very helpful.